core/src/wallet/bitcoin/api_impl/BitcoinLikeTransactionApi.cpp

```diff
@@ -268,6 +268,54 @@ namespace ledger {
             return writer.toByteArray();
         }
 
+        void BitcoinLikeTransactionApi::setSignatures(const std::vector<api::BitcoinLikeSignature> & signatures) {
+            if (signatures.size() != _inputs.size()) {
+                throw make_exception(api::ErrorCode::ILLEGAL_ARGUMENT, "DER signature length differs to current input length");
+            }
+            for (std::size_t i = 0; i < signatures.size(); ++i) {
+                BytesWriter writer;
+                //Var bytes Signature (prefix length)
+                //Get length of VarInt representing length of R
+                BytesWriter rLength;
+                rLength.writeVarInt(signatures[i].r.size());
+                //Get length of VarInt representing length of R
+                BytesWriter sLength;
+                sLength.writeVarInt(signatures[i].s.size());
+                //Get length of VarInt representing length of R and S (plus their stack sizes)
+                auto sAndRLengthInt = 1 + rLength.toByteArray().size() + signatures[i].r.size() + 
+                    1 + sLength.toByteArray().size() + signatures[i].s.size();
+                BytesWriter sAndRLength;
+                sAndRLength.writeVarInt(sAndRLengthInt);
+                //DER Signature = Total Size | DER prefix | Size(S+R) | R StackSize | R Length | R | S StackSize | S Length | S
+                auto totalSigLength = 1 + sAndRLength.toByteArray().size() + sAndRLengthInt;
+                writer.writeVarInt(totalSigLength);
+                //DER prefix
+                writer.writeByte(0x30);
+                //Size of DER signature minus DER prefix | Size(S+R)
+                writer.writeVarInt(sAndRLengthInt);
+                //R field
+                writer.writeByte(0x02); //Nb of stack elements
+                writer.writeVarInt(signatures[i].r.size());
+                writer.writeByteArray(signatures[i].r);
+                //S field
+                writer.writeByte(0x02); //Nb of stack elements
+                writer.writeVarInt(signatures[i].s.size());
+                writer.writeByteArray(signatures[i].s);
+                writer.writeByte(0x01); // SIGHASH byte
+                _inputs[i]->setP2PKHSigScript(writer.toByteArray());
+                
+            }
+        }
+
+        void BitcoinLikeTransactionApi::setDERSignatures(const std::vector<std::vector<uint8_t>> & signatures) {
+            if (signatures.size() != _inputs.size()) {
+                throw make_exception(api::ErrorCode::ILLEGAL_ARGUMENT, "DER signature length differs to current input length");
```

Comment:
s/to/from

Reply:
Also, I’m not a huge fan of exceptions. An exception means weak-typed error handling and hence, there’s no way to quickly see whether a function will “handle errors.” In the case of Live / Vault, it means an Internal Error instead of explicitly handling the returned error — plus, cross-language exception handling is rarely something easy 😛

Reply:
Look better with specific error code ?